src/main/java/com/shade/decima/ui/data/viewer/model/DMFExporter.java

```diff
@@ -0,0 +1,1468 @@
+        scene.buffers.add(buffer);
+        return buffer;
+    }
+
+    @NotNull
+    public Path getBuffersPath() throws IOException {
+        final Path buffersPath = output.resolve("dbuffers");
+        Files.createDirectories(buffersPath);
+        return buffersPath;
+    }
+
+    @Nullable
+    private FollowResult follow(@NotNull RTTIReference reference, @NotNull CoreBinary current) throws IOException {
+        final CoreBinary binary;
+        final RTTIObject uuid;
+
+        if (reference instanceof RTTIReference.External ref) {
+            final Packfile packfile = project.getPackfileManager().findFirst(ref.path());
+            if (packfile == null) {
+                throw new IOException("Couldn't find referenced file: " + ref.path());
+            }
+            binary = CoreBinary.from(packfile.extract(ref.path()), project.getTypeRegistry());
+            uuid = ref.uuid();
+        } else if (reference instanceof RTTIReference.Internal ref) {
+            binary = current;
+            uuid = ref.uuid();
+        } else {
+            return null;
+        }
+
+        final RTTIObject object = binary.find(uuid);
+
+        if (object == null) {
+            throw new IOException("Couldn't find referenced entry: " + GGUUIDValueHandler.toString(uuid));
+        }
+
+        return new FollowResult(binary, object);
+    }
+
+    @NotNull
+    private Gson createGson() {
+        return new GsonBuilder()
+            .registerTypeHierarchyAdapter(List.class, new JsonListSerializer())
+            .registerTypeHierarchyAdapter(DMFBuffer.class, new JsonBufferSerializer())
+            .registerTypeHierarchyAdapter(DMFNodeType.class, new JsonNodeTypeSerializer())
+            .disableHtmlEscaping()
+            .setPrettyPrinting()
+            .create();
+    }
+
+    private static class JsonListSerializer implements JsonSerializer<List<?>> {
+        @Override
+        public JsonElement serialize(List<?> src, Type type, JsonSerializationContext context) {
+            if (src == null || src.isEmpty()) {
+                return null;
+            }
+            final JsonArray result = new JsonArray();
+            for (Object o : src) {
+                result.add(context.serialize(o));
+            }
+            return result;
+        }
+    }
+
+    private class JsonBufferSerializer implements JsonSerializer<DMFBuffer> {
+        @Override
+        public JsonElement serialize(DMFBuffer src, Type type, JsonSerializationContext context) {
+            try {
+                return src.serialize(DMFExporter.this, context);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+    }
+
+    private class JsonNodeTypeSerializer implements JsonSerializer<DMFNodeType> {
+        @Override
+        public JsonElement serialize(DMFNodeType src, Type type, JsonSerializationContext context) {
+            return new JsonPrimitive(src.toString());
+        }
+    }
```

Comment:
Why? Gson can serialize enums. Why would you write a custom serializer for it?

Reply:
How should i go about this. Without it node type is serialized as enum member name (upper case)